### PR TITLE
Allow populating a child class of `ListView` purely from its `compose`

### PR DESCRIPTION
**Pull request by davep**
See #1588. Simply put, this change will remove the problems encountered (no default highlighted item, no item gets highlighted on focus, exception thrown when attempting to cursor through the items) when inherited from a `ListView` and then populating its items purely from `compose`.

### CODE REVIEW

**Review by rodrigogiraoserrao — APPROVED** [review]
+1 for “let's just tickle it again”

File: src/textual/widgets/_list_view.py [hunk 1]
```diff
@@ -52,6 +52,15 @@ def __init__(
...
         self.index = initial_index
 
+    def on_mount(self) -> None:
+        """Ensure the ListView is fully-settled after mounting."""
+        # If someone inherits from ListView, they might be populating it
+        # from Widget.compose rather than from passing the ListItems to the
+        # initialisation call. Given that watch_index does all the work,
+        # let's just tickle it again after the DOM has been spun up and all
+        # children of this widget will have been mounted.
+        self.index = self.index or 0
```
Comment (darrenburns) [inline src/textual/widgets/_list_view.py:62]: Small thing, but the validator would run twice I think.

I think we want to do something like `self._index = initial_index` inside the init, then in `on_mount` do `self.index = self._index`?

Comment (davep) [reply src/textual/widgets/_list_view.py:62]: I'll give that a shot.

Comment (davep) [reply src/textual/widgets/_list_view.py]: Done: 7544448002a6c382c5cd136933bbc7cf7465330d

**Review by darrenburns — APPROVED** [review]

**Pull request merged**